packages/blockchain-extension/extension/util/ExtensionUtil.ts

```diff
@@ -658,64 +658,67 @@ export class ExtensionUtil {
 
             const envEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll(true, true);
 
-            const runtimeManager: LocalEnvironmentManager = LocalEnvironmentManager.instance();
+            // If the user has no local environments, we can just update the global state automatically.
+            if (envEntries.length > 0) {
+                const runtimeManager: LocalEnvironmentManager = LocalEnvironmentManager.instance();
 
-            const runtimes: LocalEnvironment[] = [];
-            for (const entry of envEntries) {
+                const runtimes: LocalEnvironment[] = [];
+                for (const entry of envEntries) {
 
-                const localRuntime: LocalEnvironment = await runtimeManager.ensureRuntime(entry.name, undefined, entry.numberOfOrgs);
-                runtimes.push(localRuntime);
-            }
+                    const localRuntime: LocalEnvironment = await runtimeManager.ensureRuntime(entry.name, undefined, entry.numberOfOrgs);
+                    runtimes.push(localRuntime);
+                }
 
-            const generatorSemver: semver.SemVer = semver.coerce(generatorVersion); // Generator semver
-            const storedSemver: semver.SemVer = semver.coerce(extensionData.generatorVersion); // Stored generator semver
+                const generatorSemver: semver.SemVer = semver.coerce(generatorVersion); // Generator semver
+                const storedSemver: semver.SemVer = semver.coerce(extensionData.generatorVersion); // Stored generator semver
 
-            const generatorMajor: number = generatorSemver.major;
-            const storedMajor: number = storedSemver.major;
-            if (generatorMajor > storedMajor) {
-                // If major changes, then we should just force teardown without asking.
-                teardownRuntimes = true;
-            }
+                const generatorMajor: number = generatorSemver.major;
+                const storedMajor: number = storedSemver.major;
+                if (generatorMajor > storedMajor) {
+                    // If major changes, then we should just force teardown without asking.
```

Comment:
I think we need to tell the user we will tear them down and why

Reply:
If there is a major version, `teardownRuntimes = true`, then on line 685, we will log that all local runtimes will be torn down.